graphene_django/conftest.py

```diff
@@ -0,0 +1,13 @@
+import pytest
+
+from graphene_django.settings import graphene_settings
+
+from .registry import reset_global_registry
+
+
+@pytest.fixture(autouse=True)
+def reset_registry_fixture(db):
+    settings = dict(graphene_settings.__dict__)
+    yield None
+    reset_global_registry()
+    graphene_settings.__dict__ = settings
```

Comment:
Why do you have to restore the graphene_settings dict?

Reply:
There are tests that change the settings and don't change it back.

There are several options:

1. Provide a graphene_settings fixture (probably the best way). This fixture resets the settings after use

2. Just reset the settings on every test, so it is impossible to leak any settings. I decided to use this approach because it is best for an environment where contributors often change.

3. Manually reset the settings in the tests

We usually use 1. but we have regular contributors. Let me know what you prefer.

Uses of graphene_settings:

```
$> git grep graphene_settings | grep '/tests/'
graphene_django/forms/tests/test_mutation.py:from ...settings import graphene_settings
graphene_django/forms/tests/test_mutation.py:    graphene_settings.CAMELCASE_ERRORS = True
graphene_django/forms/tests/test_mutation.py:    graphene_settings.CAMELCASE_ERRORS = False
graphene_django/rest_framework/tests/test_mutation.py:from ...settings import graphene_settings
graphene_django/rest_framework/tests/test_mutation.py:    graphene_settings.CAMELCASE_ERRORS = True
graphene_django/rest_framework/tests/test_mutation.py:    graphene_settings.CAMELCASE_ERRORS = False
graphene_django/tests/test_converter.py:from ..settings import graphene_settings
graphene_django/tests/test_converter.py:    graphene_settings.DJANGO_CHOICE_FIELD_ENUM_V3_NAMING = True
graphene_django/tests/test_converter.py:    graphene_settings.DJANGO_CHOICE_FIELD_ENUM_V3_NAMING = False
graphene_django/tests/test_query.py:from ..settings import graphene_settings
graphene_django/tests/test_query.py:    graphene_settings.RELAY_CONNECTION_ENFORCE_FIRST_OR_LAST = True
graphene_django/tests/test_query.py:    graphene_settings.RELAY_CONNECTION_MAX_LIMIT = 100
graphene_django/tests/test_query.py:    graphene_settings.RELAY_CONNECTION_ENFORCE_FIRST_OR_LAST = False
graphene_django/tests/test_query.py:    graphene_settings.RELAY_CONNECTION_MAX_LIMIT = 100
graphene_django/tests/test_query.py:    graphene_settings.RELAY_CONNECTION_ENFORCE_FIRST_OR_LAST = False
graphene_django/tests/test_types.py:from ..settings import graphene_settings
graphene_django/tests/test_types.py:        graphene_settings.DJANGO_CHOICE_FIELD_ENUM_V3_NAMING = True
graphene_django/tests/test_types.py:        graphene_settings.DJANGO_CHOICE_FIELD_ENUM_V3_NAMING = False
graphene_django/tests/test_types.py:        graphene_settings.DJANGO_CHOICE_FIELD_ENUM_CUSTOM_NAME = (
graphene_django/tests/test_types.py:        graphene_settings.DJANGO_CHOICE_FIELD_ENUM_CUSTOM_NAME = None
```

Reply:
I think a fixture would be good as well. If it's going to be too much work to change it all though I'm ok leaving it like this.